src/streaming/net/FetchLoader.js

```diff
@@ -55,288 +56,291 @@ function FetchLoader(cfg) {
 
     function load(httpRequest) {
 
-        // Variables will be used in the callback functions
-        const requestStartTime = new Date();
-        const request = httpRequest.request;
-
-        const headers = new Headers(); /*jshint ignore:line*/
-        if (request.range) {
-            headers.append('Range', 'bytes=' + request.range);
-        }
+        modifyRequest(httpRequest, requestModifier)
+            .then(() => {
+                // Variables will be used in the callback functions
+                const requestStartTime = new Date();
```

Comment:
It feels a little odd that the request timer starts after the asynchronous modifications complete, but before synchronous. I would think this should be consistent.

Reply:
The idea was to keep `requestStartTime` specific to the HTTP transaction. I can see valid reasons for either approach.

Reply:
Agree with @littlespex here, prefer to do the modification before starting `load` . This would align with `modifyRequestURL`